src/timesync/timesyncd-manager.c

```diff
@@ -1109,7 +1110,9 @@ int manager_new(Manager **ret) {
 
         m->ratelimit = (RateLimit) { RATELIMIT_INTERVAL_USEC, RATELIMIT_BURST };
 
-        m->save_time_interval_usec = DEFAULT_SAVE_TIME_INTERVAL_USEC;
+        /* on systems with no RTC, default to 30s save interval */
+        m->save_time_interval_usec = dir_is_empty("/sys/class/rtc/") == true
+                        ? 30 * USEC_PER_SEC : USEC_INFINITY;
```

Comment:
please move this into a helper function `determine_default_save_time_interval_usec()` or so. Also, -ENOENT on dir_is_empty() should be treated identical to "no rtc found".

Also, please debug log if we see an unexpected error on dir_is_empty(), i.e. anything but == 0, > 0 or == -ENOENT

Reply:
Dropping this in favor of saving on all systems (see below).